File: LennardJones.f90
program lennard_jones
!!============================================================================
!! Name : Manas Mahale
!! Calculate Lennard Jones Potential
!! r       -> real
!! sigma   -> read
!! eps     -> read
!! return Lennard Jones Potential
!!============================================================================
    implicit none  
    integer :: r
    real :: sigma
    real :: eps
    real :: V_lj

    sigma = 3.38
    eps = 19.8
    read *, r
    do r = -r, r
        V_lj = 4*eps*((sigma / r)**12 - (sigma / r)**6)
        print *, 'V_lj = ', V_lj
    end do

end program lennard_jones

File: LennardJones.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

float epsilon   = 19.8; 
float sigma     = 3.38;
float r;

double lennard_jones(double r) {
    double V_lg = 4*epsilon*(pow((sigma / r), 12) - pow((sigma / r), 6));
    return V_lg;
}

int main(int argc, char *argv[]) {
    r = atof(argv[1]);
    printf("Lennard-Jones Potential\n");
    for(float i = -r; i < r; i++){
        printf("r = %f Angstrom, V_lg = %E kJ/mol\n", i, lennard_jones(i));   
    }
    return 0;   
}


